Setup ActivityPreset and UserActivity

AleksandarRistic99 opened the pull request:
closes #18
closes #21

AleksandarRistic99 committed:
Added ActivityPreset and UserActivity

Also added support classes and Repositories.

-Made UserStatus for activity --> UserDecisionStatus (mentioned in presentation to change the name)

-Made UserDecisionStatus to be an entity in a table, otherwise no relationship between UserDecisionStatus and UserActivity was possible (maybe there is another solution to enable relationship but dont store UserDecisionStatus as a entity in a table ?)

AleksandarRistic99 committed:
removed comment

AleksandarRistic99 commented:
Also added support classes and Repositories.

-Made UserStatus for activity --> UserDecisionStatus (mentioned in presentation to change the name)

-Made UserDecisionStatus to be an entity in a table, otherwise no relationship between UserDecisionStatus and UserActivity was possible (maybe there is another solution to enable relationship but dont store UserDecisionStatus as a entity in a table ?)

SamuelBruegger reviewed (approved):
- ih de class UserDecisionStatus chönt meh sich öberlege de DecisionStatus zum ne enum z mache anstatt boolean, falls meh jemols wet funktione, wie superlike oder so ihbaue abr da isch halt eifach eh design frog, wo meh au spöter glaubs guet chöntet ändere

AleksandarRistic99 commented:
> * ih de class UserDecisionStatus chönt meh sich öberlege de DecisionStatus zum ne enum z mache anstatt boolean, falls meh jemols wet funktione, wie superlike oder so ihbaue abr da isch halt eifach eh design frog, wo meh au spöter glaubs guet chöntet ändere

Guete punkt, chame denn sicher no in zuekunft beachte, eventuell en enum mache wo TRUE, FALSE, und no wiiteri optione het wie SUPERLIKE

AleksandarRistic99 committed:
Update DecisionStatus --> higher extendability

made it to be enum --> makes code extendable

AleksandarRistic99 commented:
Han de vorschlag vom Samuel mal apasst, ich denk es schadet sicher ned wenn mer de code extendable halte

Taremeh commented on src/main/java/ch/uzh/ifi/hase/soprafs21/entities/ActivityPreset.java:
```diff
@@ -0,0 +1,56 @@
+    @Column(nullable = false)
+    private ActivityCategory activityCategory;
+
+    /**
+     * googlePOI was left out, since we have to discuss how the relationship shall be between
+     * the GooglePOI object and the activityPreset --> should one activity have multiple locations, since for example football can be played on multiple locations?
+     private GooglePOI googlePOI;
+     */
```
```suggestion
     private GooglePOI googlePOI;
```

I'd say that's alright. The GooglePOI object only contains minimal information for the API request (e.g. 'keyword = sushi' for an activity _eat at sushi restaurant_). There is no location saved in `GooglePOI`.

Taremeh committed:
Update src/main/java/ch/uzh/ifi/hase/soprafs21/entities/ActivityPreset.java

Taremeh commented on src/main/java/ch/uzh/ifi/hase/soprafs21/entities/ActivityPreset.java:
```diff
@@ -0,0 +1,52 @@
+    @Column(nullable = false)
+    private ActivityCategory activityCategory;
+
+     private GooglePOI googlePOI;
```
```suggestion
    private GooglePOI googlePOI;
```

Taremeh committed:
deleted whitespace

Taremeh reviewed (changes requested):
- Good point regarding the `DecisionStatus`.
However, I would rename it to `SwipeStatus` since it's not directly clear what's meant by decision status. `SwipeStatus` would be clearer, also regarding SuperLike. 

- To simplify naming, I'd also suggest to call activities just `Activity` instead of `UserActivity`. (user activity sounds as if the user has created the activity which is not true)

- Regarding `GooglePOI`: that's all fine (see https://github.com/sopra-fs21-group-16/mth-server/pull/47#discussion_r606850667). Please implement the Spring Annotation and getters & setters. 

And btw. thx for deleting the outdated imports 🙏

AleksandarRistic99 committed:
AdaptingTareksSuggestions

-DecisionStatus --> SwipeStatus
-UserActivity --> Activity
-Spring annotation and getters & setters for GooglePOI

AleksandarRistic99 committed:
Update comment

Taremeh commented:
⚠️  Still missing getters & setters for GooglePOI (https://github.com/sopra-fs21-group-16/mth-server/pull/47#discussion_r606850667)
